replica/compaction_group.hh

```diff
@@ -173,14 +198,24 @@ public:
     const std::vector<sstables::shared_sstable>& compacted_undeleted_sstables() const noexcept;
     // Triggers regular compaction.
     void trigger_compaction();
+    bool compaction_disabled() const;
+    future<unsigned> estimate_pending_compactions() const;
 
     compaction_backlog_tracker& get_backlog_tracker();
+    void register_backlog_tracker(compaction_backlog_tracker new_backlog_tracker);
 
     size_t live_sstable_count() const noexcept;
     uint64_t live_disk_space_used() const noexcept;
     uint64_t total_disk_space_used() const noexcept;
 
-    compaction::table_state& as_table_state() const noexcept;
+    // With static sharding, i.e. vnodes, there will be only one active view.
+    compaction::compaction_group_view& as_view_for_static_sharding() const;
+    // Default view to be used on newly created sstables, e.g. those produced by repair or memtable.
+    compaction::compaction_group_view& default_view_for_new_sstables() const;
```

Comment:
SStables  produced by repair itself are repaired sstable, while the sstables produced by memtable are unrepaired.  We'd better avoid the default view, it is hard to reason about what is default.  I think it is better to use repaired_view or unrepaired_view explicitly.

Reply:
Sure, will address this.